### DISCUSSION TITLE
DWPreprocessor

### DISCUSSION THREAD

**Discussion by Nasashie** [Comfy-Org/ComfyUI#3999] [Q&A]
Hello, I came to because I have a problem with ControlNet and DWpreprocessor! I'm new to artificial intelligence and I want to extract motion estimators from a video. When I run my prompt I get an import error below, I tried to install ControlNet with Manager, then with git clone, but it's the same in both methods. I re-installed the python packaging with the cmd command, but still the same. I don't know what to do and I don't understand how to get out of this error :/

```
Error occurred when executing DWPreprocessor:

cannot import name 'packaging' from 'pkg_resources' (C:\Users\Nasa\AppData\Roaming\StabilityMatrix\Packages\ComfyUI\venv\lib\site-packages\pkg_resources\__init__.py)
 
...
```

**Comment by BrechtCorbeel** [comment]
does ... mean there is more?

can you show the contents of __init__.py in C:\Users\Nasa\AppData\Roaming\StabilityMatrix\Packages\ComfyUI\venv\lib\site-packages\pkg_resources\__init__.py?

**Comment by Nasashie** [comment]
I found the solution ! it's a setuptools version problem you have to downgrade to version 69.5.1 

`python.exe -m pip install setuptools==69.5.1` 

you have to uninstall first of course

**Reply by TTTrouble** [reply]
Thanks so much for posting this solution, fixed my problem in minutes rather than god knows how long.
